tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,173 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	runningTime := 60 * time.Second // time for which operation should be performed
+	noOfPrefixes := 20              // total number of prefixes which will be watched upon
+	watchPerPrefix := 10            // number of watchers per prefix
+	reqRate := 20                   // put request per second
+	keyPrePrefix := 1000            // max number of keyPrePrefixs for put operation
+
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	prePrefixKeys := generateRandomKeys(10, keyPrePrefix)
+	// map for holding prefixes and values
+	prefixKeysMap := make(map[string][]string)
+	// actual keys for performing put operation
+	keys := make([]string, 0)
+
+	// initialize the map with all prefixes
+	for _, prefix := range prefixes {
```

Comment:
@xiang90 we are doing exactly the same thing which u suggested using maps. Since the keys and prefixes are randomly we are pre-determining exactly what keys will get updated on which prefix and later on simply comparing with the pre-determined keys with watch events received. I am simply not able to figure out why its looking complicate to you know. Its just doing 4 simple things:
1. generate keys and prefixes.
2. For each final key put, it for all possible prefixes.
3. Put all the pre-generated keys
4. Watch on pre-determined pre-fixes and compare the values.